Microsoft.Identity.Web/TokenAcquisitionB2C.cs

```diff
@@ -0,0 +1,396 @@
+﻿using Microsoft.AspNetCore.Authentication.AzureADB2C.UI;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Web.TokenCacheProviders;
+using Microsoft.Net.Http.Headers;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Web
+{
+    public class TokenAcquisitionB2C : ITokenAcquisition
```

Comment:
a lot of duplicate code here w/the TokenAcquisition class. can you just have the stuff specific to B2C here and use the other as a base class?